clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java

```diff
@@ -386,6 +390,10 @@ public long maxOffset() {
             return maxOffset;
         }
 
+        public long minOffset() {
```

Comment:
Can we modify some of the test cases in `MemoryRecordsTest` to ensure this is getting set correctly?

It might also be useful to document what exactly the min offset is with respect to batches and records.

Reply:
Sounds good

Reply:
I've added some unresolved comments about min offsets above.